File: smash-ranks-android/app/src/main/java/com/garpr/android/misc/ThreadUtils.java
package com.garpr.android.misc;

import android.support.annotation.NonNull;

public interface ThreadUtils {

    void run(@NonNull final Task task);

    void runOnBackground(@NonNull final Runnable task);

    void runOnUi(@NonNull final Runnable task);


    interface Task {

        void onBackground();

        void onUi();

    }

}

File: smash-ranks-android/app/src/main/java/com/garpr/android/misc/ThreadUtils.kt
package com.garpr.android.misc

import androidx.annotation.UiThread
import androidx.annotation.WorkerThread
import java.util.concurrent.ExecutorService

interface ThreadUtils {

    val executorService: ExecutorService

    val isUiThread: Boolean

    fun run(task: Task)

    fun runOnBackground(task: Runnable)

    fun runOnUi(task: Runnable)

    interface Task {
        @WorkerThread
        fun onBackground()

        @UiThread
        fun onUi()
    }

}
